src/utils.js

```diff
@@ -3,6 +3,21 @@ export default class Utils {
     return Utils.escapedSchema(schema, quote) + Utils.escape((prefix || '') + table, quote);
   }
 
+  static escapeIdentifier(identifier) {
+    if (identifier == null || identifier.length === 0) {
+      return '';
+    }
+
+    const needsQuotes = /[^_A-Z0-9]/i.test(identifier) ||
```

Comment:
I'm not sure I understand what this regex is trying to do. Here's how I'm interpreting it:
If the identifier doesn't have any of _, A-Z, a-z, or 0-9 (is this right?)
OR if the identifier starts with a number. (is this right?)

Reply:
This is the exact same code that the sqldiff package uses when it generates PostgreSQL code. It just isn't visible outside the SchemaGenerator class. Hence the need to add it here.

But yes you are correct at what the regex does.

Reply:
Weird... Having trouble imagining why that's the criteria for needing quotes, but if it's what the lib does, I guess that's fine

Reply:
The logic for needing to add quotes for an identifier (like a column name) is roughly:

1.) It contains any “special characters”
2.) or, it starts with a number

This code is attempting to not over-quote identifiers because when you use quotes on a column in DDL it then _requires_ quotes always to access it, which is unexpected.